omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -479,13 +481,19 @@ protected PageableResult doSearch(RequestContext context) {
 		}
 		
 		String encounterUuid = context.getRequest().getParameter("encounter");
+		boolean includeVoided = Boolean.parseBoolean(context.getRequest().getParameter("includeAll"));
 		if (encounterUuid != null) {
 			Encounter enc = ((EncounterResource1_8) Context.getService(RestService.class).getResourceBySupportedClass(
 			    Encounter.class)).getByUniqueId(encounterUuid);
 			if (enc == null)
 				return new EmptySearchResult();
-			List<Obs> obs = new ArrayList<Obs>(enc.getAllObs());
-			return new NeedsPaging<Obs>(obs, context);
+			if (includeVoided) {
```

Comment:
Can you refactor this, in way you may not need to use the if since you have already acquired the value of includeVoided

Reply:
This has been corrected